File: WorldFeed/Server/src/WorldFeed.Common/Public/Events/QueryAwaitingEventArgs.cs
namespace WorldFeed.Common.Public.Events
{
    using System;

    using WorldFeed.Common.Models.Authentication;
    using WorldFeed.Common.Public.Models.RateLimits;

    public class QueryAwaitingEventArgs : EventArgs
    {
        private readonly string query;
        private readonly IEndpointRateLimit queryRateLimit;
        private readonly IReadOnlyTwitterCredentials twitterCredentials;

        public QueryAwaitingEventArgs(
            string query,
            IEndpointRateLimit queryRateLimit,
            IReadOnlyTwitterCredentials twitterCredentials)
        {
            this.query = query;
            this.queryRateLimit = queryRateLimit;
            this.twitterCredentials = twitterCredentials;
        }

        public string Query => query;
        public IEndpointRateLimit QueryRateLimit => queryRateLimit;
        public IReadOnlyTwitterCredentials Credentials => twitterCredentials;
        public DateTime ResetDateTime => queryRateLimit.ResetDateTime;
        public int ResetInMilliseconds => (int)queryRateLimit.ResetDateTimeInMilliseconds;
    }
}

File: Sharebook/Client/src/app/core/Public/Events/QueryAwaitingEventArgs.ts
import {IEndpointRateLimit} from '../Models/RateLimits/IEndpointRateLimit';
import {IReadOnlyTwitterCredentials} from '../../Core/Models/Authentication/ReadOnlyTwitterCredentials';
import { DateTime } from 'typescript-dotnet-commonjs/System/Time/DateTime';

export class QueryAwaitingEventArgs /*extends EventArgs */ {
  private readonly _query: string;
  private readonly _queryRateLimit: IEndpointRateLimit;
  private readonly _twitterCredentials: IReadOnlyTwitterCredentials;

  constructor(query: string, queryRateLimit: IEndpointRateLimit, twitterCredentials: IReadOnlyTwitterCredentials) {
    // super();

    this._query = query;
    this._queryRateLimit = queryRateLimit;
    this._twitterCredentials = twitterCredentials;
  }

  get query(): string {
    return this._query;
  }

  get queryRateLimit(): IEndpointRateLimit {
    return this._queryRateLimit;
  }

  get credentials(): IReadOnlyTwitterCredentials {
    return this._twitterCredentials;
  }

  get resetDateTime(): DateTime /*DateTimeOffset*/ {
    return this._queryRateLimit.resetDateTime;
  }

  get resetInMilliseconds(): number {
    return this._queryRateLimit.resetDateTimeInMilliseconds as number;
  }
}
